x/concentrated-liquidity/tick.go

```diff
@@ -410,22 +416,37 @@ func (k Keeper) GetTickLiquidityNetInDirection(ctx sdk.Context, poolId uint64, t
 	for ; iterator.Valid(); iterator.Next() {
 		tickIndex, err := types.TickIndexFromBytes(iterator.Key())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		tickStruct, err := ParseTickFromBz(iterator.Value())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		liquidityDepth := queryproto.TickLiquidityNet{
 			LiquidityNet: tickStruct.LiquidityNet,
 			TickIndex:    tickIndex,
 		}
+
+		if zeroForOne {
+			if liquidityDepth.TickIndex < currentTick {
+				if liquidityDepth.LiquidityNet.IsNegative() {
+					startTickLiquidity = startTickLiquidity.Sub(liquidityDepth.LiquidityNet)
+				}
+			}
+		} else {
+			if liquidityDepth.TickIndex > currentTick {
+				if liquidityDepth.LiquidityNet.IsPositive() {
+					startTickLiquidity = startTickLiquidity.Add(liquidityDepth.LiquidityNet)
+				}
```

Comment:
We should be able to always add. I don't understand why this is a conditional

Reply:
I also don't understand why tests aren't failing. Do we have a test case going over 2 positions? Same for zero for one